cpp/include/cudf/utilities/type_dispatcher.hpp

```diff
@@ -82,6 +82,39 @@ struct id_to_type_impl {
 template <cudf::type_id Id>
 using id_to_type = typename id_to_type_impl<Id>::type;
 
+/**
+ * @brief "Returns" the corresponding type that is stored on the device when using `cudf::column`
+ *
+ * Use this "type function" with the `using` type alias:
+ * @code
+ * using Type = get_column_stored_type<Element>;
+ * @endcode
+ *
+ * @tparam T The literal type that is stored on the host
+ */
+// clang-format off
+template <typename T>
+using get_column_stored_type =
+  std::conditional_t<std::is_same<numeric::decimal32, T>::value, int32_t,
+  std::conditional_t<std::is_same<numeric::decimal64, T>::value, int64_t, T>>;
```

Comment:
I think I like this nomenclature better. Or maybe `storage_type_t`. 
```suggestion
using underlying_type_t =
  std::conditional_t<std::is_same<numeric::decimal32, T>::value, int32_t,
  std::conditional_t<std::is_same<numeric::decimal64, T>::value, int64_t, T>>;
```

Reply:
Can we keep the word `column` in it? Maybe `underlying_column_type_t`?

Reply:
`column_storage_type_t` is actually probably the best.

Reply:
Done.